### DISCUSSION TITLE
Clustering in Artery

### DISCUSSION THREAD

**Discussion by amicha24** [riebl/artery#242] [Q&A] [answered]
I am trying to implement some clustering algorithm(s) in artery for my master thesis and I was wondering if there is any work that is related to what I am trying to achieve where the authors use artery. (I saw a number that used Veins and I am trying to connect the dots.)

If you have any suggestions on the classes I should have a more extensive look at, I would be grateful.

I am thinking about the local environment and a cam sensor but I have not yet tried anything.

Thank you

**Comment by riebl** [comment, marked as answer]
I am unaware of any clustering and/or platooning algorithms realised in Artery. Fellows at TU Delft use Artery for V2X communication but implement their platooning algorithms directly in their OpenTrafficSim (OTS). If you use SUMO as traffic simulator, you can issue any TraCI command by a vehicle's V2X service and thus form vehicle clusters.

**Reply by amicha24** [reply]
Ok, thank you for the answer!

If I may also ask, the way I see it, I am trying to create a new, custom service. Would you agree, or would you say it is unnecessary?

**Reply by riebl** [reply]
Yes, I suggest creating another service module for this purpose.

**Reply by amicha24** [reply]
Thank you1
